From af2241c678bcd9af58d38f98f5caec96a498733e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fatih=20A=C5=9F=C4=B1c=C4=B1?= <fatih.asici@gmail.com>
Date: Fri, 23 May 2014 17:02:01 +0300
Subject: [PATCH] QDnsLookup: Fix build with uClibc

uClibc doesn't have a nsmap member in __res_state. Since it also doesn't have
res_nquery() which is mandatory for QDnsLookup, we can simply disable the code
to fix the build.

Sent-Upstream: https://codereview.qt-project.org/86159
Signed-off-by: Fatih Aşıcı <fatih.asici@gmail.com>
Change-Id: Ia872f535519aca3a2de763548c6dd0e3e0ee20d4
---
 src/network/kernel/qdnslookup_unix.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/network/kernel/qdnslookup_unix.cpp b/src/network/kernel/qdnslookup_unix.cpp
index 26834df..2a482a6 100644
--- a/src/network/kernel/qdnslookup_unix.cpp
+++ b/src/network/kernel/qdnslookup_unix.cpp
@@ -160,9 +160,11 @@ void QDnsLookupRunnable::query(const int requestType, const QByteArray &requestN
                 Q_CHECK_PTR(ns);
                 state._u._ext.nsaddrs[0] = ns;
             }
+#ifndef __UCLIBC__
             // Set nsmap[] to indicate that nsaddrs[0] is an IPv6 address
             // See: https://sourceware.org/ml/libc-hacker/2002-05/msg00035.html
             state._u._ext.nsmap[0] = MAXNS + 1;
+#endif
             state._u._ext.nscount6 = 1;
             ns->sin6_family = AF_INET6;
             ns->sin6_port = htons(53);
-- 
1.9.1

